clippy_lints/src/doc/markdown.rs

```diff
@@ -117,6 +134,17 @@ fn check_word(cx: &LateContext<'_>, word: &str, span: Span, code_level: isize, b
         // try to get around the fact that `foo::bar` parses as a valid URL
         && !url.cannot_be_a_base()
     {
+        let Some(fragment_span) = fragments.span(cx, range.clone()) else {
+            return ControlFlow::Break(());
+        };
+
+        let span = Span::new(
+            fragment_span.lo() + BytePos::from_usize(fragment_offset),
+            fragment_span.lo() + BytePos::from_usize(fragment_offset + word.len()),
+            fragment_span.ctxt(),
+            fragment_span.parent(),
+        );
+
```

Comment:
Should you not be adjusting the range before creating the span? `fragment_offset` looks like it's an offset in the markdown text.

Reply:
I'm not sure if I understand this comment correctly. This snippet is taken as-is from `check`  with variable names fixed, `check->offset` didn't really care about the markdown text.

Reply:
`fragment_offset` is an offset in the cooked doc string. It can't be used as an offset for a span since that doesn't always line up perfectly with the source text.

Reply:
After testing this out, `text_to_check` only contains text, it doesn't contain links, or bold text, etc. And `fragment_offset` is resetted for each one of those `text`s. I can add a debug assertion for future proofing this though.

Reply:
A text fragment can still contain escape sequences e.g. `#[doc = "docs with unicode \u{xxxxxx}"]`. The string the fragments work on is the cooked version of the doc string, not the source form. Multiline comments (`/** */`) might also have issues, don't know how the those are presented.

Reply:
Just tested this, `#[doc = "*"]` does not lint, at all. Even for the test cases documented, if you change the `/// *` for `#[doc = "*"]`  they will just not lint.

For the case of `/** XXX */`, seems that everything works correctly (or that I'm testing for the wrong thing), either way, I've added some tests for this along with some other weird escaped sequences.

Reply:
Don't really want to stall the PR on this and it's not a new thing added. I'll try to make it break later.